recipes/libnghttp3/all/conanfile.py

```diff
@@ -0,0 +1,88 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.apple import is_apple_os
+from conan.tools.cmake import CMake, CMakeToolchain, CMakeDeps, cmake_layout
+from conan.tools.files import get, rmdir, copy
+from conan.tools.gnu import PkgConfigDeps
+from conan.tools.microsoft import is_msvc
+from conan.tools.scm import Version
+import os
+
+
+required_conan_version = ">=1.53.0"
+
+
+class Nghttp3Conan(ConanFile):
+    name = "libnghttp3"
+    description = "HTTP/2 C Library and tools"
+    topics = ("http", "http3")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://nghttp2.org/nghttp3/"
+    license = "MIT"
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    def build_requirements(self):
+        self.tool_requires("cmake/[>=3.20 <4]")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def validate(self):
+        if self.settings.compiler == "gcc" and Version(self.settings.compiler.version) < "6":
+            raise ConanInvalidConfiguration(f"{self.ref} requires GCC >= 6.0")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["ENABLE_SHARED_LIB"] = self.options.shared
+        tc.variables["ENABLE_STATIC_LIB"] = not self.options.shared
+        tc.variables["ENABLE_LIB_ONLY"] = True
+        if is_apple_os(self):
+            # workaround for: install TARGETS given no BUNDLE DESTINATION for MACOSX_BUNDLE executable
+            tc.cache_variables["CMAKE_MACOSX_BUNDLE"] = False
+        tc.generate()
+
+        tc = CMakeDeps(self)
+        tc.generate()
+        tc = PkgConfigDeps(self)
+        tc.generate()
+
```

Comment:
```suggestion
```
I see no requirement listed. The tool requirement for cmake should be solved by VirtualBuildEnv

Reply:
Could you please elaborate on this comment? I don't understand how that applies to the code suggestion.

For CMake recipes, as I understand it, it is common to run the CMakeToolchain generator as well as the CMakeDeps generator. This project also has pkg-config support and that generator is run as well.

Reply:
I believe I've addressed this correctly. Let me know if I have not.